launchdarkly-android-client-sdk/src/main/java/com/launchdarkly/sdk/android/ContextDataManager.java

```diff
@@ -49,62 +49,55 @@ final class ContextDataManager {
     private final CopyOnWriteArrayList<LDAllFlagsListener> allFlagsListeners =
             new CopyOnWriteArrayList<>();
     private final LDLogger logger;
-    private final Object writerLock = new Object();
+
+    /**
+     * This lock is to protect context, flag, and persistence operations.
+     */
+    private final Object lock = new Object();
 
     @NonNull private volatile LDContext currentContext;
     @NonNull private volatile EnvironmentData flags = new EnvironmentData();
     @NonNull private volatile ContextIndex index = null;
-    private volatile String flagsContextId = null;
 
     ContextDataManager(
             @NonNull ClientContext clientContext,
             @NonNull PersistentDataStoreWrapper.PerEnvironmentData environmentStore,
             int maxCachedContexts
     ) {
-        this.currentContext = clientContext.getEvaluationContext();
         this.environmentStore = environmentStore;
         this.maxCachedContexts = maxCachedContexts;
         this.taskExecutor = ClientContextImpl.get(clientContext).getTaskExecutor();
         this.logger = clientContext.getBaseLogger();
+        switchToContext(clientContext.getEvaluationContext());
     }
 
     /**
-     * Returns the current context.
+     * Switches to providing flag data for the provided context.
```

Comment:
For reviewers: You can no longer switch context without switching the flags this DataManager will use.  Weird that was ever an option to begin with...